File: python/hash_file.py
from __future__ import absolute_import, division, print_function, unicode_literals

# A command-line program that computes the Dropbox-Content-Hash of the given file.

import sys

from dropbox_content_hasher import DropboxContentHasher

def main():
    prog_name, args = sys.argv[0], sys.argv[1:]
    if len(args) != 1:
        sys.stderr.write("Expecting exactly one argument, got {}.\n".format(len(args)))
        sys.exit(1)

    fn = args[0]

    hasher = DropboxContentHasher()
    with open(fn, 'rb') as f:
        while True:
            chunk = f.read(1024)  # or whatever chunk size you want
            if len(chunk) == 0:
                break
            hasher.update(chunk)
    print(hasher.hexdigest())

if __name__ == '__main__':
    main()

File: rust/src/hash_file.rs
//! A command-line tool that computes the Dropbox-Content-Hash of the given file.

extern crate digest;
extern crate dropbox_content_hasher;

use std::io::Write as io_Write;
use std::io::Read as io_Read;
use digest::Digest;
use dropbox_content_hasher::DropboxContentHasher;

fn main() {
    let mut args = std::env::args();
    args.next().unwrap();  // Remove name of binary.
    if args.len() != 1 {
        writeln!(&mut std::io::stderr(), "Expecting exactly one argument, got {}.", args.len()).unwrap();
        std::process::exit(1);
    }

    let file_name = args.next().unwrap();

    let mut hasher = DropboxContentHasher::new();
    let mut buf: [u8; 4096] = [0; 4096];
    let mut f = std::fs::File::open(file_name).unwrap();
    loop {
        let len = f.read(&mut buf).unwrap();
        if len == 0 { break; }
        hasher.input(&buf[..len])
    }
    drop(f);

    let hex_hash = format!("{:x}", hasher.result());
    println!("{}", hex_hash);
}